qiskit/algorithms/time_evolvers/trotterization/trotter_qrte.py

```diff
@@ -132,44 +136,84 @@ def evolve(self, evolution_problem: TimeEvolutionProblem) -> TimeEvolutionResult
             ValueError: If an unsupported Hamiltonian type is provided.
         """
         evolution_problem.validate_params()
-        if evolution_problem.t_param is not None:
-            raise ValueError(
-                "TrotterQRTE does not accept a time dependent Hamiltonian,"
-                "``t_param`` from the ``TimeEvolutionProblem`` should be set to ``None``."
-            )
 
         if evolution_problem.aux_operators is not None and self.estimator is None:
             raise ValueError(
                 "The time evolution problem contained ``aux_operators`` but no estimator was "
                 "provided. The algorithm continues without calculating these quantities. "
             )
         hamiltonian = evolution_problem.hamiltonian
-        if not isinstance(hamiltonian, (Pauli, PauliSumOp)):
+        if not isinstance(hamiltonian, (Pauli, PauliSumOp, SparsePauliOp)):
             raise ValueError(
                 f"TrotterQRTE only accepts Pauli | PauliSumOp, {type(hamiltonian)} provided."
             )
-        # the evolution gate
-        evolution_gate = PauliEvolutionGate(
-            hamiltonian, evolution_problem.time, synthesis=self.product_formula
-        )
+        if evolution_problem.t_param is not None and len(_get_parameters(hamiltonian.coeffs)) > 1:
+            raise ValueError("Time-dependent Hamiltonian cannot contain multiple parameters")
+
+        # make sure PauliEvolutionGate does not implement more than one Trotter step
+        dt = evolution_problem.time / self.num_timesteps
+        if self.product_formula.settings["reps"] != 1 and self.product_formula.settings["order"] == 1:
+            self.product_formula = LieTrotter(
+                order=self.product_formula.settings["order"],
+                reps=1,
+                insert_barriers=self.product_formula.settings["insert_barriers"],
+                cx_structure=self.product_formula.settings["cx_structure"]
+            )
+        elif self.product_formula.settings["reps"] != 1 and self.product_formula.settings["order"] > 1:
+            self.product_formula = SuzukiTrotter(
+                order=self.product_formula.settings["order"],
+                reps=1,
+                insert_barriers=self.product_formula.settings["insert_barriers"],
+                cx_structure=self.product_formula.settings["cx_structure"]
+            )
```

Comment:
Yes, the code works with `reps>1`. The problem here is that this means the time step is implicitly reduced without telling the user. `time=1, num_timesteps=10, reps=10` will result in a product formula with `dt=0.01` instead of `dt=0.1` as expected from `num_timesteps`.

Reply:
As long as we document that as desired behavior I think it's fine 🙂 Decoupling this also allows the user a more fine-grained control over the accuracy.